pkg/cmd/provisioning/utils.go

```diff
@@ -23,3 +27,34 @@ func EnsureDir(path string) error {
 
 	return nil
 }
+
+// CreateShellScript creates a shell script given commands to execute
+func CreateShellScript(commands []string) string {
+	return fmt.Sprintf("#!/bin/sh\n%s", strings.Join(commands, "\n"))
+}
+
+// SaveToFile saves the given data to a given file
+func SaveToFile(filePurpose, filePath string, data []byte) error {
+	log.Printf("Saving %s locally at %s", filePurpose, filePath)
+	f, err := os.Create(filePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create file %s to store %s", filePath, filePurpose)
+	}
+	_, err = f.Write(data)
+	f.Close()
+	if err != nil {
+		return errors.Wrapf(err, "failed to write %s to file %s", filePurpose, filePath)
+	}
+	return nil
+}
+
+// CountNonDirectoryFiles counts files which are not a directory
+func CountNonDirectoryFiles(files []os.FileInfo) int {
```

Comment:
Is this used outside of the test files?

Reply:
that's fair. let each test file at least share this functionality :+1:

Reply:
fixed